src/command_modules/azure-cli-cdn/azure/cli/command_modules/cdn/custom.py

```diff
@@ -0,0 +1,129 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import azure.cli.core.azlogging as azlogging
+from azure.mgmt.cdn.models import (Endpoint, QueryStringCachingBehavior, SkuName,
+                                   EndpointUpdateParameters, ProfileUpdateParameters)
+
+
+def list_profiles(client, resource_group_name=None):
+    profiles = client.profiles
+    profile_list = profiles.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else profiles.list()
+    return list(profile_list)
+
+
+def _update_mapper(existing, new, keys):
```

Comment:
In the future this would probably be worth extracting out to core.utils or something.